### PR TITLE
Minor fixes to `ActiveSpaceTransformer`

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This fixes some minor bugs in the `ActiveSpaceTransformer`.

### CODE REVIEW

File: qiskit_nature/transformers/second_quantization/electronic/active_space_transformer.py [hunk 1]
```diff
@@ -363,28 +364,16 @@ def _transform_property(self, prop: Property) -> Property:
         Raises:
             TypeError: if an unexpected Property subtype is encountered.
         """
-        transformed_property: Property
+        transformed_property: Property = None
```
Comment (woodsp-ibm) [inline qiskit_nature/transformers/second_quantization/electronic/active_space_transformer.py:367]: I would think this instance var ought to be annotated as Optional now

File: releasenotes/notes/fix-as-trafo-grouped-property-iteration-bca2096817b1e497.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    Some minor issues in the :class:`~qiskit_nature.transformers.second_quantization.electronic.ActiveSpaceTransformer`
+    were fixed. Namely:
+
+    - iterated properties are no longer deep-copied resulting in non-transformed ones actually being dropped rather than carried on
```
Comment (woodsp-ibm) [inline releasenotes/notes/fix-as-trafo-grouped-property-iteration-bca2096817b1e497.yaml:7]: Can a simple unit test be done to assert the behavior now is/remains as its now expected.

Comment (mrossinek) [reply releasenotes/notes/fix-as-trafo-grouped-property-iteration-bca2096817b1e497.yaml:7]: On second thought after my initial "thumb up": adding such a test now would be slightly artificial because the `ASTransformer` actually supports mapping all currently existing properties. This fix is a "backport" from the #640 PR because it also removes some other edge cases in the iteration behavior. I will check again tomorrow with Julian if we can create a unittest for one of those edge cases without needing the Seniority Zero transformer :+1:

Comment (mrossinek) [reply releasenotes/notes/fix-as-trafo-grouped-property-iteration-bca2096817b1e497.yaml:7]: I added a simple test. It asserts that the `ElectronicBasisTransform` gets updated correctly which it did not prior to this PR (i.e. said test will fail on current `main`). This is does not cover the entirety of this PRs behavioral change but should be sufficient until we have a transformer which can drop certain properties during the transformation.

**Review by pbark — APPROVED** [review]
LGTM! Thanks

**Review by woodsp-ibm — APPROVED** [review]
